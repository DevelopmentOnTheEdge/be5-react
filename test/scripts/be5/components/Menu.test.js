import React    from 'react';
import renderer from 'react-test-renderer';
import Menu     from  '../../../../src/scripts/be5/components/menu/Menu';
import NavbarMenu from  '../../../../src/scripts/be5/components/menu/NavbarMenu';
import {getTestUser} from "../testUtils";


it('snapshot menu', () => {
  const menu = {"root":[{"action":{"arg":"static/welcome.be","name":"call"},"default":false,"title":"Добро пожаловать"},{"children":[{"action":{"arg":"login","name":"call"},"default":false,"title":"Вход"}],"default":false,"id":{"_en_":"users"},"title":"Пользователи"},{"children":[{"action":{"arg":"table/welfareGroups/Муниципальные услуги","name":"call"},"default":false,"title":"Муниципальные услуги"}],"default":false,"id":{"_en_":"_welfareGroups_"},"title":"Услуги"}]}
  // be5.net.request = function (path, attr, callback) {
  //   callback()
  // };

  const component = renderer.create(
    <Menu currentRoles={["Guest"]} menu={menu} fetchMenu={()=>{}}/>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('snapshot NavbarMenu', () => {
  const navbarMenu = {"root":[{"action":{"arg":"static/welcome.be","name":"call"},"default":true,"id":{"_en_":"_welcome_","_qn_":"All records"},"title":"Добро пожаловать"},{"children":[{"action":{"arg":"table/testtable/Test 1D","name":"call"},"default":false,"id":{"_en_":"testtable","_qn_":"Test 1D"},"title":"Test 1D"},{"action":{"arg":"table/testtable/Test 1D unknown","name":"call"},"default":false,"id":{"_en_":"testtable","_qn_":"Test 1D unknown"},"title":"Test 1D unknown"}],"default":false,"id":{"_en_":"testtable"},"title":"Test table"},{"action":{"arg":"table/users/All records","name":"call"},"default":false,"id":{"_en_":"users","_qn_":"All records"},"operations":[{"action":{"arg":"form/users/All records/Insert","name":"call"},"id":{"_en_":"users","_on_":"Insert"},"title":"Добавить"}],"title":"Пользователи"},{"action":{"arg":"logout","name":"call"},"default":false,"id":{"_en_":"_logout_","_qn_":"All records"},"title":"Выход"}]}

  //for prevent error - Tooltip cannot find ID
  const div = document.createElement('div');
  div.setAttribute("id", "RoleSelector");
  document.body.appendChild(div);

  const component = renderer.create(
    <NavbarMenu user={getTestUser()} menu={navbarMenu} fetchMenu={()=>{}} toggleRoles={()=>{}}/>
  );

  expect(component.toJSON()).toMatchSnapshot();

  // changeDocument('MAIN_DOCUMENT', { component: StaticPage, value: "test" });
  // expect(component.toJSON()).toMatchSnapshot();

});
