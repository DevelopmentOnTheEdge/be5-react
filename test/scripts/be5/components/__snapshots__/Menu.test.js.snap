// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot NavbarMenu 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#!static/welcome.be"
            onClick={[Function]}
          >
            Добро пожаловать
          </a>
        </li>
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            href="#"
            onClick={[Function]}
          >
            Test table
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu"
            role="menu"
            tabIndex="-1"
          >
            <a
              className="dropdown-item"
              href="#!table/testtable/Test 1D"
              onClick={[Function]}
              tabIndex="0"
              type={undefined}
            >
              Test 1D
            </a>
            <a
              className="dropdown-item"
              href="#!table/testtable/Test 1D unknown"
              onClick={[Function]}
              tabIndex="0"
              type={undefined}
            >
              Test 1D unknown
            </a>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#!table/users/All records"
            onClick={[Function]}
          >
            Пользователи
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#!logout"
            onClick={[Function]}
          >
            Выход
          </a>
        </li>
      </ul>
      <form
        className="form-inline ml-auto"
      >
        <div
          className="roleBox mr-sm-2 dropdown"
          id="RoleSelector"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            Roles
          </button>
          <div
            aria-hidden={true}
            className="dropdown-menu"
            role="menu"
            tabIndex="-1"
          >
            <div
              className="role"
            >
              <input
                checked={false}
                id="Administrator-checkbox"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="Administrator-checkbox"
              >
                <span
                  className="checkBox"
                />
                Administrator
              </label>
            </div>
            <div
              className="role"
            >
              <input
                checked={true}
                id="Manager-checkbox"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="Manager-checkbox"
              >
                <span
                  className="checkBox"
                />
                Manager
              </label>
            </div>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
              type={undefined}
            />
            <div
              className="roleBox_add-actions"
            >
              Выбрать:
               
              <button
                className="enable-all btn btn-primary btn-sm"
                onClick={[Function]}
                type="button"
              >
                Всё
              </button>
               
              <button
                className="disable-all btn btn-secondary btn-sm"
                onClick={[Function]}
                type="button"
              >
                Ничего
              </button>
            </div>
          </div>
        </div>
         
        <a
          className="btn btn-secondary"
          href="#!logout"
          onClick={[Function]}
          type={undefined}
        >
          Logout
        </a>
      </form>
    </div>
  </div>
</nav>
`;

exports[`snapshot menu 1`] = `
<div
  className="menuContainer"
>
  <input
    className="searchField form-control"
    onChange={[Function]}
    placeholder="Filter..."
    type="text"
    value=""
  />
  <div
    className="menu"
  >
    <div
      className="menuNode"
    >
      <a
        className="rootMenuItem menuItemWithRef"
        href="#!static/welcome.be"
        onClick={[Function]}
        target=""
      >
        Добро пожаловать
      </a>
      <div />
    </div>
    <div
      className="menuNode"
    >
      <span
        className="rootMenuItem menuItemWithoutRef"
      >
        Пользователи
      </span>
      <div />
      <ul>
        <li>
          <div
            className="menuNode"
          >
            <a
              className="menuItem menuItemWithRef"
              href="#!login"
              onClick={[Function]}
              target=""
            >
              Вход
            </a>
            <div />
          </div>
        </li>
      </ul>
    </div>
    <div
      className="menuNode"
    >
      <span
        className="rootMenuItem menuItemWithoutRef"
      >
        Услуги
      </span>
      <div />
      <ul>
        <li>
          <div
            className="menuNode"
          >
            <a
              className="menuItem menuItemWithRef"
              href="#!table/welfareGroups/Муниципальные услуги"
              onClick={[Function]}
              target=""
            >
              Муниципальные услуги
            </a>
            <div />
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
